src/Emanote/Model/Note.hs

```diff
@@ -98,9 +98,11 @@ lookupMeta k =
   lookupAeson Nothing k . _noteMeta
 
 noteTitle :: Note -> Text
-noteTitle Note {..} =
-  fromMaybe (R.routeBaseName . R.lmlRouteCase $ _noteRoute) $
-    getPandocTitle _noteDoc
+noteTitle note@Note {..} =
+  let yamlNoteTitle = lookupMeta (one "title") note
+      fileNameTitle = R.routeBaseName . R.lmlRouteCase $ _noteRoute
+      notePandocTitle = getPandocTitle _noteDoc
+   in fromMaybe (fromMaybe fileNameTitle notePandocTitle) yamlNoteTitle
```

Comment:
By the way, you can use [`Alternative`](https://hackage.haskell.org/package/base-4.15.0.0/docs/Control-Applicative.html#t:Alternative) to refactor this into,

```haskell
fromMaybe fileNameTitle $ yamlNoteTitle <|> notePandocTitle 
```

Reply:
cool, learned that today!